util/mqtt.c

```diff
@@ -355,6 +355,28 @@ mqtt_init (const char *server_uri)
   return 0;
 }
 
+/**
+ * @brief Reinitializes communication after mqtt_reset was used
+ *
+ */
+void
+mqtt_reinit ()
```

Comment:
I would not do it in a separate function for this case. Else I think its getting a bit confusing.
We already have mqtt_reset and mqtt_init.

Reply:
I would rather rename it then, but as it is indeed a re-initialization for the current fork after a mqtt_reset was called. Also it is documented as that

Reply:
I don't like that function because it does not really reinit. It first checks if it needs to reinit then reinits or does nothing.
So in 99% of cases this functions just does nothing.
If we really want to have a separate function I would do the `if ((mqtt_get_global_client ()) == NULL)` outside of this function.